src/gpu/intel/jit/gemm/include/type.hpp

```diff
@@ -99,13 +99,13 @@ class Type {
     }
     constexpr Type baseType() const { return *this; }
 
-    template <typename U> constexpr friend int operator*(U a, Type t) {
-        return t.isInt4() ? int((unsigned(a) + 1) >> 1) : int(a * (U(1) << t.log2Size()));
+    template <typename U> constexpr friend U operator*(U a, Type t) {
```

Comment:
Also, I'm a bit afraid that `U` might be a small type (say uint8_t) and we'll get unexpected wraparounds. Let's use the usual C++ type deduction rules for multiplication by an int:
```suggestion
    template <typename U> constexpr friend decltype(std::declval<U>() * 1) operator*(U a, Type t) {
```